hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieCompactionConfig.java

```diff
@@ -99,6 +99,12 @@
       .withDocumentation("Controls how compaction scheduling is triggered, by time or num delta commits or combination of both. "
           + "Valid options: " + Arrays.stream(CompactionTriggerStrategy.values()).map(Enum::name).collect(Collectors.joining(",")));
 
+  public static final ConfigProperty<String> ASYNC_COMPACT_ENABLE = ConfigProperty
```

Comment:
We should check how this new config affects hthe default behavior of async compaction in `HoodieDeltaStreamer`  and also whether it is in sync with `DataSourceWriteOptions`.

Reply:
Good point.  I'll check this.

Reply:
I removed this config and added a Kafka-Connect specific config for async compaction: `ASYNC_COMPACT_ENABLE` (default is `true`).  The async compaction is enabled only when this config is true and MOR table is used for writing.